x-pack/legacy/plugins/task_manager/task_manager.ts

```diff
@@ -99,63 +130,87 @@ export class TaskManager {
       definitions: this.definitions,
       taskManagerId: `kibana:${taskManagerId}`,
     });
+    // pipe store events into the TaskManager's event stream
+    this.store.events.subscribe(event => this.events$.next(event));
 
-    const pool = new TaskPool({
+    this.pool = new TaskPool({
       logger: this.logger,
       maxWorkers: this.maxWorkers,
     });
-    const createRunner = (instance: ConcreteTaskInstance) =>
-      new TaskManagerRunner({
-        logger: this.logger,
-        instance,
-        store,
-        definitions: this.definitions,
-        beforeRun: this.middleware.beforeRun,
-        beforeMarkRunning: this.middleware.beforeMarkRunning,
-      });
-    const poller = new TaskPoller<FillPoolResult>({
-      logger: this.logger,
+
+    this.claimRequests$ = new Subject();
+    this.pollingSubscription = Subscription.EMPTY;
```

Comment:
Yeah, TBH I avoided it because I noticed a pattern throughout Kibana where all properties are initialised in the constructor....

But moved some stuff around now.